packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.types.ts

```diff
@@ -369,3 +359,23 @@ export interface UsePickerValuePrivateContextValue {
    */
   dismissViews: () => void;
 }
+
+export interface SetValueActionOptions<TError = string> {
+  /**
+   * Importance of the change when picking a value:
+   * - "accept": fires `onChange`, fires `onAccept` and closes the picker.
+   * - "set": fires `onChange` but do not fire `onAccept` and does not close the picker.
+   * @default "accept"
+   */
+  changeImportance?: PickerChangeImportance;
+  /**
+   * The validation error associated to the current value.
+   * If not defined, the validation will be re-applied by the picker.
+   */
+  validationError?: TError;
+  /**
+   * The shortcut that triggered this change.
+   * Should not be defined if the change does not come from a shortcut.
+   */
+  shortcut?: PickersShortcutsItemContext;
```

Comment:
In the Base UI version we won't have a notion of shortcut, so this field will probably only live in the MUI version (since some people needed it).

The other will be handled by the Base UI layer.